code/game/objects/items/devices/scanners.dm

```diff
@@ -823,3 +823,133 @@ GENE SCANNER
+	else 
+		icon_state = "extrapolator_scan"
+		scan = TRUE
+		to_chat(user, "<span class='notice'>You put the probe back in the device and set it to SCAN</span>")
+
+/obj/item/extrapolator/examine(mob/user)
+	. = ..()
+	if(in_range(user, src) || isobserver(user))
+		if(!scanner)
+			. += "<span class='notice'>The scanner is missing.<span>"
+		else
+			. += "<span class='notice'>A class <b>[scanner.rating]</b> scanning module is installed. It is <i>screwed</i> in place.<span>"
+		if(cooldown > world.time - (1200 / scanner.rating))
+			. += "<span class='warning'>The extrapolator is still recharging!</span>"
+		else
+			. += "<span class='info'>The extrapolator is ready to use!</span>"
+	
+
+/obj/item/extrapolator/attack(atom/AM, mob/living/user)
+	return 
+
+/obj/item/extrapolator/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(scanner)
+		if(!target.extrapolator_act(user, src, scan))
+			if(locate(/datum/component/infective) in target.datum_components)
+				return //so the failure message does not show when we can actually extrapolate from a component
+			if(scan)
+				to_chat(user, "<span class='notice'>the extrapolator fails to return any data</span>")
+			else
+				to_chat(user, "<span class='notice'>the extrapolator's probe detects no diseases</span>")
+	else 
+		to_chat(user, "<span class='warning'>the extrapolator has no scanner installed</span>")
+
+/obj/item/extrapolator/proc/scan(atom/AM, var/list/diseases = list(), mob/user)
+	to_chat(user, "<span class='notice'><b>[src] detects the following diseases:</b></span>")
+	for(var/datum/disease/D in diseases)
+		if(istype(D, /datum/disease/advance))
+			var/datum/disease/advance/A = D
+			if(A.properties["stealth"] >= (2 + scanner.rating)) //the extrapolator can detect diseases of higher stealth than a normal scanner
+				continue
+			to_chat(user, "<span class='info'><font color='green'><b>[A.name]</b>, stage [A.stage]/5</color></span>")
+			to_chat(user, "<span class='info'><b>[A] has the following symptoms:</b></span>")
+			for(var/datum/symptom/S in A.symptoms)
+				to_chat(user, "<span class='info'>[S.name]</span>")
+		else
+			to_chat(user, "<span class='info'><font color='green'><b>[D.name]</b>, stage [D.stage]/[D.max_stages].</color></span>")
+
+/obj/item/extrapolator/proc/extrapolate(atom/AM, var/list/diseases = list(), mob/user, isolate = FALSE, timer = 200)
+	var/list/advancediseases = list()
+	var/list/symptoms = list()
+	if(cooldown > world.time - (1200 / scanner.rating))
+		to_chat(user, "<span class='warning'>The extrapolator is still recharging!</span>")
+		return
+	for(var/datum/disease/advance/cantidate in diseases)
+		advancediseases += cantidate
+	if(!advancediseases.len)
+		to_chat(user, "<span class='warning'>There are no valid diseases to make a culture from.</span>")
+	var/datum/disease/advance/A = input(user,"What disease do you wish to extract") in null|advancediseases

```

Comment:
Not sure how this is meant to work, but when testing this input never actually showed up, even if there were 2 diseases on an object

Reply:
Was it with a normal disease or an advanced disease? Extract doesnt work on non-advanced diseases

Reply:
yep, works fine when tested